contracts/cw1-subkeys/src/contract.rs

```diff
@@ -505,714 +651,1060 @@ mod tests {
+            execute(
+                deps.as_mut(),
+                mock_env(),
+                info,
+                ExecuteMsg::Execute { msgs },
+            )
+            .unwrap_err();
+
+            assert_eq!(
+                query_all_allowances(deps.as_ref(), None, None)
+                    .unwrap()
+                    .canonical(),
+                AllAllowancesResponse { allowances: vec![] }.canonical()
+            );
+        }
 
-        let initial_allowances = vec![coin(amount1, denom1), coin(amount2, denom2)];
+        #[test]
+        fn not_enough_allowance() {
+            let Suite { mut deps, .. } = SuiteConfig::new()
+                .with_allowance(SPENDER1, coin(10, TOKEN1))
+                .init();
 
-        let expires_height = Expiration::AtHeight(5432);
-        let expires_never = Expiration::Never {};
-        // Initially set first spender allowance with height expiration, the second with no expiration
-        let initial_expirations = vec![expires_height, expires_never];
+            let msgs = vec![BankMsg::Send {
+                to_address: SPENDER2.to_owned(),
+                amount: coins(20, TOKEN1),
+            }
+            .into()];
 
-        let info = mock_info(owner, &[]);
-        setup_test_case(
-            deps.as_mut(),
-            &info,
-            &admins,
-            &initial_spenders,
-            &initial_allowances,
-            &initial_expirations,
-        );
+            let info = mock_info(SPENDER1, &[]);
 
-        // Subtract from spender1 (existing) account (has none of that denom)
-        let msg = ExecuteMsg::DecreaseAllowance {
-            spender: spender1.to_string(),
-            amount: allow3,
-            expires: None,
-        };
-        let res = execute(deps.as_mut(), mock_env(), info.clone(), msg);
+            execute(
+                deps.as_mut(),
+                mock_env(),
+                info,
+                ExecuteMsg::Execute { msgs },
+            )
+            .unwrap_err();
+
+            assert_eq!(
+                query_all_allowances(deps.as_ref(), None, None)
+                    .unwrap()
+                    .canonical(),
+                AllAllowancesResponse {
+                    allowances: vec![AllowanceInfo {
+                        spender: SPENDER1.to_owned(),
+                        balance: NativeBalance(coins(10, TOKEN1)),
+                        expires: Expiration::Never {}
+                    }]
+                }
+                .canonical()
+            );
+        }
 
-        // Verify
-        assert!(res.is_err());
-        // Verify everything stays the same for that spender
-        let allowance = query_allowance(deps.as_ref(), spender1.to_string()).unwrap();
-        assert_eq!(
-            allowance,
-            Allowance {
-                balance: NativeBalance(vec![allow1.clone(), allow2.clone()]),
-                expires: expires_height,
+        #[test]
+        #[ignore]
```

Comment:
why ignore? does this fail?

Reply:
Yes it fails. I am pretty sure this test is correct (but double check if I understand logic), and it should be implemented, and then test would be "unignored".